jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/config/JettyWebSocketClientConfiguration.java

```diff
@@ -0,0 +1,83 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.websocket.client.config;
+
+import java.util.ServiceLoader;
+
+import org.eclipse.jetty.util.Loader;
+import org.eclipse.jetty.webapp.AbstractConfiguration;
+import org.eclipse.jetty.webapp.Configuration;
+import org.eclipse.jetty.webapp.FragmentConfiguration;
+import org.eclipse.jetty.webapp.MetaInfConfiguration;
+import org.eclipse.jetty.webapp.WebAppConfiguration;
+import org.eclipse.jetty.webapp.WebInfConfiguration;
+import org.eclipse.jetty.webapp.WebXmlConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>Websocket Configuration</p>
+ * <p>This configuration configures the WebAppContext server/system classes to
+ * be able to see the {@code org.eclipse.jetty.websocket.client} package.
+ * This class is defined in the webapp package, as it implements the {@link Configuration} interface,
+ * which is unknown to the websocket package.  However, the corresponding {@link ServiceLoader}
+ * resource is defined in the websocket package, so that this configuration only be
+ * loaded if the jetty-websocket jars are on the classpath.
+ * </p>
+ */
+public class JettyWebSocketClientConfiguration extends AbstractConfiguration
+{
+    private static final Logger LOG = LoggerFactory.getLogger(JettyWebSocketClientConfiguration.class);
+
+    public JettyWebSocketClientConfiguration()
+    {
+        addDependencies(WebXmlConfiguration.class, MetaInfConfiguration.class, WebInfConfiguration.class, FragmentConfiguration.class);
+
+        if (isAvailable("org.eclipse.jetty.osgi.annotations.AnnotationConfiguration"))
```

Comment:
I don't think you need to do this availability check. Just add the dependency as a string and it will be ordered if it is there and ignored if it is not.

Reply:
But the logic is only adding the osgi one if it is available, otherwise only adding the standard one. Wouldn't it have a different behaviour if we add both.

Reply:
But I don't think you need to do that even.  The OSGi configuration replaces the non OSGI configuration, so it should inherit all of it's dependencies and orderings.  You can't list out all the infinite possible extensions of org.eclipse.jetty.annotations.AnnotationConfiguration here, so don't try.

Reply:
hmmm for some reason the OSGi AnnotationConfiguration does not "replace" the normal AnnotationConfiguration???
Might have to check with @janbartel why?
But ultimately here you should just depend on org.eclipse.jetty.osgi.annotations.AnnotationConfiguration and the replacement mechanism should make the sorting work correctly

Reply:
Perhaps I just forgot to add it when I did the conversion.   Can you try adding to the OSGi AnnotationConfiguration:
```java

    @Override
    public Class<? extends Configuration> replaces()
    {
        return org.eclipse.jetty.annotations.AnnotationConfiguration.class;
    }
```

Reply:
I overrided the `replaces()` method in the osgi AnnotationConfiguration now. It passed the tests even without that, even after I made the changes you asked for removing the `isAvailable` checks.